web3/middleware/normalize_errors.py

```diff
@@ -0,0 +1,22 @@
+from cytoolz import (
+    assoc,
+    dissoc,
+)
+
+
+def normalize_errors_middleware(make_request, web3):
+    def middleware(method, params):
+        result = make_request(method, params)
+        if method == 'eth_getTransactionReceipt' and 'error' in result:
```

Comment:
This block probably deserves a comment.  Code reads pretty well, but a nice comment explaining why will be unambiguous.